src/s2python/s2_parser.py

```diff
@@ -0,0 +1,91 @@
+import json
+import logging
+from typing import Optional, TypeVar
+
+from s2python.validate_values_mixin import ValidateValuesMixin
+from s2python.s2_validation_error import S2ValidationError
+from s2python.common import *
+from s2python.frbc import *
+
+LOGGER = logging.getLogger(__name__)
+MessageType = str
+S2Message = ValidateValuesMixin
+
+C = TypeVar("C", bound=S2Message)
+
+
+# May be generated with development_utilities/generate_s2_message_type_to_class.py
+TYPE_TO_MESSAGE_CLASS = {
+    "FRBC.ActuatorStatus": FRBCActuatorStatus,
+    "FRBC.FillLevelTargetProfile": FRBCFillLevelTargetProfile,
+    "FRBC.Instruction": FRBCInstruction,
+    "FRBC.LeakageBehaviour": FRBCLeakageBehaviour,
+    "FRBC.StorageStatus": FRBCStorageStatus,
+    "FRBC.SystemDescription": FRBCSystemDescription,
+    "FRBC.TimerStatus": FRBCTimerStatus,
+    "FRBC.UsageForecast": FRBCUsageForecast,
+    "Handshake": Handshake,
+    "HandshakeResponse": HandshakeResponse,
+    "InstructionStatusUpdate": InstructionStatusUpdate,
+    "PowerForecast": PowerForecast,
+    "PowerMeasurement": PowerMeasurement,
+    "ReceptionStatus": ReceptionStatus,
+    "ResourceManagerDetails": ResourceManagerDetails,
+    "RevokeObject": RevokeObject,
+    "SelectControlType": SelectControlType,
+    "SessionRequest": SessionRequest,
+}
+
+
+class S2Parser:
```

Comment:
Maybe, we could have a method that accepts both a str and a dict:

```python
def parse(self, message: str | dict) -> S2Message:
    if isintance(message, str):
        message = json.loads(message)
    
    message_type = self.get_message_type_from_dict(message)

    if message_type not in TYPE_TO_MESSAGE_CLASS:
        raise S2ValidationError(
            message,
            f"Unable to parse {message_type} as an S2 message. Type unknown.",
        )

    return TYPE_TO_MESSAGE_CLASS[message_type].parse_obj(message)    
```

Reply:
Fixed!